src/pudl/io_managers.py

```diff
@@ -683,6 +685,47 @@ class FercXBRLSQLiteIOManager(FercSQLiteIOManager):
     metadata.
     """
 
+    @staticmethod
+    def use_latest_filing_for_context(
+        table: pd.DataFrame, unique_cols: list[str]
+    ) -> pd.DataFrame:
+        """Get facts from the latest filing that reported for each context.
+
+        We treat two XBRL contexts that are the same except for their IDs as the same context.
+        """
+        strict_deduped = table.drop_duplicates()
+        logger.debug(
+            f"Dropped {len(table) - len(strict_deduped)} completely duplicated rows."
+        )
+        chrono_order = table.sort_values("publication_time")
```

Comment:
> I'm not sure it makes a functional difference since the next `drop_duplicates()` is more liberal, but Is this supposed to be `strict_deduped.sort_values()` rather than `table.sort_values()`? It doesn't look like `strict_deduped` is ever referred to again after it is set above, other than to log the number of dropped records.

Yep, thanks for catching!

> It seems like it would be possible to condense all of these iterative `drop_duplicates()` into a single sort-and-drop_dupes. We're just doing in stages so we can log and have a sense of how many duplicates we're getting at each stage?

Yeah, I was doing that to log and understand the duplicate counts. We had wanted this at some point. I turned this into a loop to be a little clearer about what's going on, but I also think it's reasonable to just run it all in one stage and log its drop rate instead of one per step. LMK if you have a preference either way.

Reply:
I don't have strong feelings either way. Just wanted to make sure I understood that this was an iterative narrowing, and that there wasn't something tricky going on that meant it had to be done in several stages.